File: PYTHON/Concesionario/gestorAplicacion/Activos/TransaccionTaller.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)
from transaccion import Transaccion
class TransaccionTaller(Transaccion):
    transaccionestal = []

    def __init__(self, tipo, ingreso, cliente, auto, articulo, mecanico, transfer):
        super().__init__(tipo, ingreso, cliente, transfer)
        self.auto = auto
        self.mecanico = mecanico
        self.articulo = articulo
        TransaccionTaller.transaccionestal.append(self)

    def info(self):
        txt = f"Transacción #{self.transfer:08d}: {self.tipo} realizado por {self.mecanico.nombre} para el cliente {self.cliente.nombre} por un total de ${self.ingreso}"
        return txt

    @staticmethod
    def get_clientePorCedula(cedula):
        finder = None
        cli = None

        for trans in TransaccionTaller.transaccionestal:
            if trans.get_clienteCed() == cedula:
                finder = trans
                cli = finder.get_cliente()
                break

        return cli

    @staticmethod
    def get_transaccionporCedula(cedula):
        finder = None

        for trans in TransaccionTaller.transaccionestal:
            if trans.get_clienteCed() == cedula:
                finder = trans.get_auto()
                break

        return finder

    def set_auto(self, auto):
        self.auto = auto

    def get_auto(self):
        return self.auto

    def set_mecanico(self, mecanico):
        self.mecanico = mecanico

    def get_mecanico(self):
        return self.mecanico

    def set_articulo(self, articulo):
        self.articulo = articulo

    def get_articulo(self):
        return self.articulo

    @staticmethod
    def agregar_transaccion(transaccion):
        TransaccionTaller.transaccionestal.append(transaccion)

    @staticmethod
    def eliminar_transaccion(transaccion):
        TransaccionTaller.transaccionestal.remove(transaccion)

    @staticmethod
    def get_transaccionestal():
        return TransaccionTaller.transaccionestal
    @staticmethod
    def set_transaccionestal(pp):
        TransaccionTaller.transaccionestal=pp

File: Concesionario/src/gestorAplicacion/activos/TransaccionTaller.java
package gestorAplicacion.activos;

import java.io.Serializable;
import java.util.ArrayList;

import gestorAplicacion.personal.*;

public class TransaccionTaller extends Transaccion implements Serializable{
	private static final long serialVersionUID = 1L;
	private Auto auto;
	private Mecanico mecanico;
	private Articulo articulo;
	static ArrayList<TransaccionTaller> transaccionestal = new ArrayList<TransaccionTaller>();
	
	public TransaccionTaller(String tipo, long ingreso, Cliente cliente, Auto auto,Articulo articulo, Mecanico mecanico, int transfer) {
		super(tipo, ingreso, cliente, transfer);
		this.auto=auto;
		this.mecanico=mecanico;
		this.articulo=articulo;
		TransaccionTaller.transaccionestal.add(this);
	}
	@Override
	public String info() {
		String txt = String.format("Transacción #%08d: %s realizado por %s para el cliente %s por un total de $%d",
	            transfer, getTipo(), mecanico.getNombre(), cliente.getNombre(), getIngreso());
			
		
		return txt;
	}
	public static Cliente getClientePorCedula(long cedula){
		Transaccion finder = null;
		Cliente cli=null;
		
		for(Transaccion trans: transaccionestal) {
			if(trans.getClienteCed()== cedula) {
				finder = trans;
				cli=finder.getCliente();
				break;
			}
		}
		return cli;
	}
	public static Auto getTransaccionporCedula(long cedula) {
		Auto finder = null;
		for (TransaccionTaller trans: transaccionestal) {
			if(trans.getClienteCed()== cedula) {
				finder = trans.getAuto();
				break;
			}
		}
		return finder;
	}
	public void setAuto(Auto auto) {
	    this.auto = auto;
	}

	public Auto getAuto() {
	    return auto;
	}
	public void setMecanico(Mecanico mecanico) {
	    this.mecanico = mecanico;
	}

	public Mecanico getMecanico() {
	    return mecanico;
	}
	public void setArticulo(Articulo articulo) {
	    this.articulo = articulo;
	}

	public Articulo getArticulo() {
	    return articulo;
	}
	public static void agregarTransaccion(TransaccionTaller transaccion) {
	    transaccionestal.add(transaccion);
	}

	public static void eliminarTransaccion(TransaccionTaller transaccion) {
	    transaccionestal.remove(transaccion);
	}

	public static ArrayList<TransaccionTaller> getTransaccionestal() {
	    return transaccionestal;
	}
}



